lib/runtime-core/src/backing.rs

```diff
@@ -103,6 +103,104 @@ impl LocalBacking {
         })
     }
 
+    /// Resets the `LocalBacking` (`Memories` and `Globals`) for an `Instance` using the provided `ModuleInfo`
+    pub(crate) fn reset(&self, module_info: &ModuleInfo) -> RuntimeResult<()> {
+        Self::reset_memories(&module_info, &self.memories)?;
+        Self::reset_globals(&module_info, &self.globals)
+    }
+
+    fn reset_memories(
+        module_info: &ModuleInfo,
+        memories: &SliceMap<LocalMemoryIndex, Memory>,
+    ) -> RuntimeResult<()> {
+        Self::shrink_memories(memories)?;
+        Self::zero_memories(memories);
+        Self::reinitialize_memories(module_info, memories)
+    }
+
+    fn shrink_memories(memories: &SliceMap<LocalMemoryIndex, Memory>) -> RuntimeResult<()> {
+        for (_index, memory) in memories.iter() {
+            memory.shrink_to_minimum()?;
+        }
+        Ok(())
+    }
+
+    fn zero_memories(memories: &SliceMap<LocalMemoryIndex, Memory>) {
+        for (_index, memory) in memories.iter() {
+            let view = &memory.view::<u64>();
```

Comment:
Will this work if allocated memory is not a multiple of 8 bytes? Or is it always multiple of 8?

Reply:
internally `view::<T>()` only uses `T` to do `(n * Page) / mem::size_of::<T>()`
`Page = 2^16 bytes`, so no issues here